packages/edit-site/src/utils/history.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { createBrowserHistory } from 'history';
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+const history = createBrowserHistory();
```

Comment:
I believe, _in theory_, this PR should work even inside iframes. If for some reasons url transitions are completely off the table, we can still switch this to `createMemoryHistory` and get the similar effect.